doc/doxygen/headers/laoperators.h

```diff
@@ -65,7 +65,42 @@
  *
  * For objects of type LinearOperator, all vector space operations, i.e.,
  * addition and subtraction, scalar multiplication and composition (of
- * compatible linear operators) are implemented.
+ * compatible linear operators) are implemented:
+ * @code
+ * dealii::LinearOperator<> op_a, op_b;
+ * double k;
+ *
+ * // vector space addition, subtraction and scalar multiplication
+ * op_a + op_b;
+ * op_a - op_b;
+ * k * op_a;
+ * op_a * k;
+ *
+ * // in-place variants
+ * op_a += op_b;
+ * op_a -= op_b;
+ * op_a *= k;
+ *
+ * // operator composition
+ * op_a * op_b;
+ * op_a *= op_b; // If op_b is an endomorphism of the domain space of op_a
+ * @endcode
+ *
+ * Further, by adding (or subtracting) a vector of the range space to
+ * (from) a linear operator, an affine variant is created:
+ * @code
+ * dealii::LinearOperator<> op_a;
+ * dealii::Vector<double> offset;
+ *
+ * op_a + offset;
+ * op_a + offset;
+ * offset - op_a;
+ * offset - op_a;
```

Comment:
I'm not sure I'm so enthusiastic about these affine variants. How do they interact with the other operations? For example, what does
```
  (op + vec) * k
```
yield? How about
```
  (op1 + vec) * op2
```
? Does this yield something different than
```
  op1 * (op2 + vec)
```
?